### PR TITLE
Add recentering to PeripheralRingMeshGenerator

### LINKED ISSUE #22138: Allow PeripheralRingMeshGenerator to Re-center the Input Mesh
## Reason
`PeripheralRingMeshGenerator` takes the input mesh as is. If the input mesh's centroid is off the origin, `PeripheralRingMeshGenerator`  will still create a peripheral ring with its center at the origin for the input, which is needed in some cases. However, sometimes it is preferred that the the input mesh can be re-centered to its centroid before creating the peripheral ring.

## Design
Add a new input parameter to `PeripheralRingMeshGenerator` to allow users to decide with re-centering is needed. Add corresponding test cases.

## Impact
Make `PeripheralRingMeshGenerator` more flexible to use.

### PR DESCRIPTION
**Pull request by miaoyinb**
closes #22138

### CODE REVIEW

File: modules/reactor/src/meshgenerators/PeripheralRingMeshGenerator.C [hunk 1]
```diff
@@ -161,11 +166,17 @@ PeripheralRingMeshGenerator::generate()
   if (*(input_mesh->elem_dimensions().begin()) != 2 ||
       *(input_mesh->elem_dimensions().rbegin()) != 2)
     paramError("input", "Only 2D meshes are supported.");
+  // Move the centroid of the mesh to (0, 0, 0) if input centroid is enforced to be the ring center.
+  const Point input_mesh_centroid = MooseMeshUtils::meshCentroidCalculator(*input_mesh);
+  if (_force_input_centroid_as_center)
+    MeshTools::Modification::translate(
+        *input_mesh, -input_mesh_centroid(0), -input_mesh_centroid(1), -input_mesh_centroid(2));
```
Comment (GiudGiud) [inline modules/reactor/src/meshgenerators/PeripheralRingMeshGenerator.C:173]: dont you need to retranslate back after too?

Comment (miaoyinb) [reply modules/reactor/src/meshgenerators/PeripheralRingMeshGenerator.C:173]: I don't think we need to translate it back.

Comment (GiudGiud) [reply modules/reactor/src/meshgenerators/PeripheralRingMeshGenerator.C:173]: so the peripheral mesh is created around 0,0,0 always, it s just a matter of whether the mesh should be centered there too?

Comment (miaoyinb) [reply modules/reactor/src/meshgenerators/PeripheralRingMeshGenerator.C:173]: Yes. It is just a matter whether to center the input mesh if it is not originally centered.

File: modules/reactor/src/meshgenerators/PeripheralRingMeshGenerator.C [hunk 2]
```diff
@@ -24,6 +24,10 @@ PeripheralRingMeshGenerator::validParams()
 {
   InputParameters params = PolygonMeshGeneratorBase::validParams();
   params.addRequiredParam<MeshGeneratorName>("input", "The input mesh to be modified.");
+  params.addParam<bool>("force_input_centroid_as_center",
+                        false,
+                        "Whether to enforce use of the centroid position of the input mesh as the "
+                        "center of the peripheral ring.");
```
Comment (GiudGiud) [inline modules/reactor/src/meshgenerators/PeripheralRingMeshGenerator.C:30]:
```suggestion
                        "Whether to enforce use of the centroid position of the input mesh as the "
                        "center of the peripheral ring by translating the input mesh to the origin.");

```

**Review by GiudGiud — COMMENTED** [review]
just a suggestion
this looks good

**Review by GiudGiud — COMMENTED** [review]
squash update commit please so we can merge

**Comment by GiudGiud** [comment]
we disabled squash merges at the repo level otherwise I would have done that

**Comment by miaoyinb** [comment]
Done

**Review by GiudGiud — APPROVED** [review]
will merge when tests are done

looking more at the extruder PR btw, I think it ll be ready as well

**Pull request merged**